actions/build.action.ts

```diff
@@ -78,72 +78,99 @@ export class BuildAction extends AbstractAction {
       (option) => option.name === 'config',
     )!.value as string;
     const configuration = await this.loader.load(configFileName);
-    const appName = commandInputs.find((input) => input.name === 'app')!
-      .value as string;
 
-    const pathToTsconfig = getTscConfigPath(
-      configuration,
-      commandOptions,
-      appName,
-    );
-    const { options: tsOptions } =
-      this.tsConfigProvider.getByConfigFilename(pathToTsconfig);
-    const outDir = tsOptions.outDir || defaultOutDir;
+    const buildAll = commandOptions.find((opt) => opt.name === 'all')!
+      .value as boolean;
 
-    const isWebpackEnabled = getValueOrDefault<boolean>(
-      configuration,
-      'compilerOptions.webpack',
-      appName,
-      'webpack',
-      commandOptions,
-    );
-    const builder = isWebpackEnabled
-      ? { type: 'webpack' }
-      : getBuilder(configuration, commandOptions, appName);
+    let appNames: string[];
+    if (buildAll) {
+      appNames = [undefined!]; // always include the default project
```

Comment:
I can change that, but it requires to also change a lot of other APIs that do accept `undefined` to mean "default app", but do not declare that.

I was reluctant to do it, since it's a lot of changes, but if it's okay I'll change those to `string | undefined`.

Reply:
Fixed this spot and all the APIs it trickles down to in https://github.com/nestjs/nest-cli/pull/2312/commits/f574cfdec963dbf760b3ec5fdd60e5f4ae099dc4.

Haven't looked at other places that accept an `appName` - they may also need to be changed to explicitly allow `undefined`, but that would be outside of this PR's scope.